### DISCUSSION TITLE
Allow configuration of React hooks rules

### DISCUSSION THREAD

**Discussion by arendjr** [rome/tools#3877] [Suggestions]
I was just looking into your implementation of the exhaustive deps rule for React hooks. It looks like a nice implementation, looking forward to it!

One thing I have been missing with the ESLint rules provided by the React team is the ability to configure additional (custom) hooks that I know will return stable results or which should also have their dependencies checked.

A common example would be for Redux’s `useDispatch()` hook, for which it is very annoying if `dispatch` needs to be needlessly specified as a dependency everywhere.

Would you consider allowing the built-in configuration to be extended in Rome?

**Comment by ematipico** [comment]
This rule has been designed to accept a configuration where a user can add additional "stable" - that's how the plugin calls them - hooks. Although we haven't documented it yet. We will do it once we are closer to having a decent implementation of the rule, there are still some missing bits.

**Reply by arendjr** [reply]
Great to hear!
